### PR TITLE
Make public key list sort of compressed public key dictionary order.

### LINKED ISSUE #61: Sorting rule of public keys.
In current implementation, public keys of signers are sorted internally.
https://github.com/chaintope/tapyrus-signer/blob/master/src/signer_node/node_parameters.rs#L38

Sorting PublicKey is based on concatenating of (x-coordinate || y-coordinate) of the point, where each coordinate is decoded as hex string with little endian.

On the other hand, a public key in Tapyrus is usually encoded as big endian and prefixed '02' or '03'.

it is better to change the rule so that public keys are sorted based on 'compressed public key' because current implementation is not intuitive

### PR DESCRIPTION
**Pull request by rantan**
Fix #61

### CODE REVIEW

File: src/signer_node/message_processor/process_candidateblock.rs [hunk 1]
```diff
@@ -194,7 +194,7 @@ mod tests {
     /// * The node's status is Member and its index is 4.
     /// * The round master's index is 0.
     ///
-    /// 1. Send candidateblock message from index 0 node(array index is 1).
+    /// 1. Send candidateblock message from index 0 node(array index is 4).
```
Comment (Yamaguchi) [inline src/signer_node/message_processor/process_candidateblock.rs]: The differences between "index" and "array index" confused me.
I think this test case does not need TEST_KEY.pubkeys().
Using sorted public keys makes the purpose of the test case clear.

File: src/signer_node/mod.rs [hunk 2]
```diff
@@ -643,53 +643,6 @@ mod tests {
...
-        let private_key = TEST_KEYS.key[0];
-        let to_address = address(&private_key);
-
-        let params = NodeParameters::new(
-            to_address,
-            pubkey_list.clone(),
-            private_key,
-            threshold,
-            MockRpc {
-                return_block: safety_error("Not set block.".to_string()),
```
Comment (Yamaguchi) [inline src/signer_node/mod.rs]: After this change, `cargo test` report the warning like this:

```
warning: unused import: `safety_error`
   --> src/signer_node/mod.rs:525:37
    |
525 |     use crate::rpc::tests::{safety, safety_error, MockRpc};
    |                                     ^^^^^^^^^^^^
    |
    = note: `#[warn(unused_imports)]` on by default

```

File: src/signer_node/node_parameters.rs [hunk 3]
```diff
@@ -65,4 +66,53 @@ impl<T: TapyrusApi> NodeParameters<T> {
             share_count: n.clone(),
         }
     }
+
+    fn sort_publickey(pubkeys: &mut Vec<PublicKey>) {
+        pubkeys.sort_by_key(|p| {
+            let serialized = p.key.serialize();
+            let hex = hex::encode(&serialized[..]);
+            hex
```
Comment (Yamaguchi) [inline src/signer_node/node_parameters.rs]: Is encoding required?
Does returning [u8] or Vec<u8> occur any problems?

File: src/signer_node/message_processor/process_candidateblock.rs (hunk 1 above)
Comment (rantan) [reply src/signer_node/message_processor/process_candidateblock.rs]: How about changing the order of TEST_KEYS to dictionary order?
This kind of confusion should occur in other test cases but we need to change some other test cases too.

File: src/signer_node/mod.rs (hunk 2 above)
Comment (rantan) [reply src/signer_node/mod.rs]: Thanks, I'll fix it.

File: src/signer_node/node_parameters.rs [hunk 4]
```diff
@@ -65,4 +66,53 @@ impl<T: TapyrusApi> NodeParameters<T> {
             share_count: n.clone(),
         }
     }
+
+    fn sort_publickey(pubkeys: &mut Vec<PublicKey>) {
```
[the remark this answers is not in the document]
Comment (rantan) [reply src/signer_node/node_parameters.rs]: I think we don't use this function in out of tapyrus-signer.
Do you have any use cases?

File: src/signer_node/node_parameters.rs (hunk 3 above)
Comment (rantan) [reply src/signer_node/node_parameters.rs]: I think it's true. I will try it.

File: src/signer_node/message_processor/process_candidateblock.rs (hunk 1 above)
Comment (rantan) [reply src/signer_node/message_processor/process_candidateblock.rs]: if it is ok, I want to create an another PR for it.

File: src/signer_node/node_parameters.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Yamaguchi) [reply src/signer_node/node_parameters.rs]: We have no use case for now.

But for example, in Lighting Network, some script use public keys sorted in lexicographical order
https://github.com/lightningnetwork/lightning-rfc/blob/master/03-transactions.md#funding-transaction-output

File: src/signer_node/message_processor/process_candidateblock.rs (hunk 1 above)
Comment (Yamaguchi) [reply src/signer_node/message_processor/process_candidateblock.rs]: > if it is ok, I want to create an another PR for it.
Ok.

File: src/signer_node/node_parameters.rs (hunk 3 above)
Comment (rantan) [reply src/signer_node/node_parameters.rs]: updated #7b4fccd

File: src/signer_node/mod.rs (hunk 2 above)
Comment (rantan) [reply src/signer_node/mod.rs]: updated 8b22c1c66b30947f07262a75c9a6da8a874a8ced

**Pull request merged**